iconservice/icx/issue/issue_regulator.py

```diff
@@ -0,0 +1,139 @@
+
+    def __init__(self):
+        self._regulator_storage: 'RegulatorStorage' = None
+
+    def open(self, context_db: 'ContextDatabase'):
+        self.close()
+        self._regulator_storage = RegulatorStorage(context_db)
+
+    def close(self):
+        """Close resources
+        """
+        if self._regulator_storage:
+            self._regulator_storage.close(context=None)
+            self._regulator_storage = None
+
+    @staticmethod
+    def _reflect_difference_in_issuing(icx_issue_amount: int, over_issued_icx: int) -> Tuple[int, int, int]:
+        assert icx_issue_amount >= 0
+
+        corrected_icx_issue_amount = icx_issue_amount - over_issued_icx
+        if over_issued_icx < 0:
+            deducted_icx = over_issued_icx
+            remain_over_issued_icx = 0
+            return deducted_icx, remain_over_issued_icx, corrected_icx_issue_amount
+
+        if corrected_icx_issue_amount >= 0:
+            remain_over_issued_icx = 0
+            deducted_icx = over_issued_icx
+        else:
+            remain_over_issued_icx = abs(corrected_icx_issue_amount)
+            corrected_icx_issue_amount = 0
+            deducted_icx = icx_issue_amount
+
+        return deducted_icx, remain_over_issued_icx, corrected_icx_issue_amount
+
+    def _get_difference_between_icon_service_and_reward_calc(self, icx_amount, i_score_amount) -> int:
+        diff = icx_amount * self._I_SCORE_EXCHANGE_RATE - i_score_amount
+        return diff
+
+    def _accumulate_current_period_issued_icx(self, context: 'IconScoreContext', issue_amount: int) -> int:
+        current_calc_period_issued_amount = self._regulator_storage.get_current_calc_period_issued_icx(context)
+        current_calc_period_issued_amount += issue_amount
+        return current_calc_period_issued_amount
+
+    def _separate_icx_and_i_score(self, i_score: int) -> Tuple[int, int]:
+        over_issued_icx = abs(i_score) // self._I_SCORE_EXCHANGE_RATE
+        over_issued_i_score = abs(i_score) % self._I_SCORE_EXCHANGE_RATE
+        if i_score < 0:
+            over_issued_icx = -over_issued_icx
+            over_issued_i_score = -over_issued_i_score
+
+        return over_issued_icx, over_issued_i_score
+        # TODO: second
+        # max_i_score = self._MAX_I_SCORE if total_over_issued_i_score >= 0 else -self._MAX_I_SCORE
+        # over_issued_icx = total_over_issued_i_score // max_i_score
+        # over_issued_icx = -over_issued_icx if max_i_score < 0 else over_issued_icx
+        # over_issued_i_score = total_over_issued_i_score % max_i_score
+
+    def correct_issue_amount_on_calc_period(self,
+                                            context: 'IconScoreContext',
+                                            prev_calc_period_issued_i_score: Optional[int],
+                                            icx_issue_amount: int) -> Tuple[int, int, int]:
+        assert icx_issue_amount >= 0
+
+        current_calc_period_total_issued_icx: int = \
```

Comment:
I want this logic divide some part like "get variable from storage -> logic -> put variable to storage"
because I was able to seek how many variable get from storage in advance.

Reply:
I have reflected your suggestion.